server.js

```diff
@@ -20,4 +20,57 @@ app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
+app.get('/messages', function (request, response) {
+  response.json(messages)
+})
+
+app.post('/messages', function (request, response) {
+  let newMessage = {};
+
+  if (request.body.from.trim() && request.body.text.trim()) {
+    newMessage.id = messages.length
+    newMessage.from = request.body.from
+    newMessage.text = request.body.text
+    newMessage.timeSent = new Date()
+    messages.push(newMessage)
+    response.json(messages)
+  } else {
+    response.status(400).json("Please fill all fileds")
+  }
+
+})
+
+app.get('/messages/search', function (request, response) {
+  let searchWord = request.query.text
+  let result = messages.filter(item => item.text.toLowerCase().includes(searchWord.toLowerCase()))
+  if (result.length > 0) {
+    response.json(result)
+  } else {
+    response.status(404).json("Nothing was found")
```

Comment:
This is a good response message. I think that a correct response code would be 204 No Content for an empty result set. 404 would be used if you had asked for a specific resource by ID and it did not exist.

Reply:
Implemented with 204